account-kit/react/src/hooks/useSolanaTransaction.ts

```diff
@@ -0,0 +1,109 @@
+import * as solanaNetwork from "../solanaNetwork.js";
+import { useMutation } from "@tanstack/react-query";
+import { SolanaSigner } from "@account-kit/signer";
+import type { BaseHookMutationArgs } from "../types.js";
+import { PublicKey, SystemProgram } from "@solana/web3.js";
+import { useSolanaSigner } from "./useSolanaSigner.js";
+import { useContext } from "react";
+import { AlchemySolanaWeb3Context } from "../AlchemySolanaWeb3Context.js";
+/**
+ * We wanted to make sure that this will be using the same useMutation that the
+ * useSendUserOperation does.
+ * We are going to flatten it to make sure that we are abstracting it, and that we have
+ * the flattened version here for readability.
+ *
+ * @see {@link useSendUserOperation}
+ * @see {@link https://tanstack.com/query/v5/docs/framework/react/reference/useMutation | TanStack Query useMutation}
+ */
+export interface SolanaTransaction {
+  /** The solana signer used to send the transaction */
+  readonly signer: SolanaSigner | void;
+  /** The solana connection used to send the transaction */
+  readonly connection: solanaNetwork.Connection | void;
+  /** The result of the transaction */
+  readonly data: void | { hash: string };
+  /** Is the use sending a transaction */
+  readonly isPending: boolean;
+  /** The error that occurred */
+  readonly error: Error | null;
+  reset(): void;
+  /** Send the transaction */
+  mutate(): void;
+  /** Send the transaction asynchronously */
+  mutateAsync(): Promise<{ hash: string }>;
+}
+
+/**
+ * The parameters for the useSolanaTransaction hook.
+ */
+export type SolanaTransactionParams = {
+  transaction: {
+    amount: number;
+    toAddress: string;
+  };
+  signer?: SolanaSigner;
+  connection?: solanaNetwork.Connection;
+  policyId?: string;
+  /**
+   * Override the default mutation options
+   *
+   * @see {@link BaseHookMutationArgs}
+   */
+  mutation?: BaseHookMutationArgs<{ hash: string }>;
+};
+
+/**
+ * This is the hook that will be used to send a transaction.
+ *
+ * @example
+ * ```ts
+  useSolanaTransaction({
+    policyId: "<policyId>",
+    transaction: {
```

Comment:
Here similarly to the useSolanaSignMessage function, the tax should be an input to the returned mutation functions rather than the hook. PolicyId seems reasonable to be an input to the hook though

Reply:
Ok, going to move both as a param. Also, going to make the param either 
{transfer: {toAddress, amount}}

And
{instructions: <TransactionInstructions[]>}
(Since this is the lower level and would let us transition into calling fn's and other